client/my-sites/stats/stats-module/index.jsx

```diff
@@ -70,6 +88,25 @@ class StatsModule extends Component {
 			// eslint-disable-next-line react/no-did-update-set-state
 			this.setState( { loaded: false } );
 		}
+
+		if ( ! isEqual( this.props.data, prevProps.data ) ) {
+			const diffData = this.calculateDiff( prevProps.data, this.props.data );
+			// eslint-disable-next-line react/no-did-update-set-state
+			this.setState( { diffData } );
+		}
+	}
+
+	calculateDiff( prevData, newData ) {
+		// TODO: Review diff logic.
+		// Is it good enough to use the first data set as a baseline?
```

Comment:
If it's a new day then we should start afresh for the base I guess.

Reply:
Something like this:

https://github.com/Automattic/wp-calypso/pull/99419/files#diff-41c000373062e93978606a49642f8c6e936f79f8590d294cc5279deb9c900b66R83


If the backend determines it's a new day, then it'll start fresh with data. Otherwise we shouldn't calculate deltas because they are from different days.

Reply:
Thanks for explaining this further in our chat today. I get the reasoning now and will look into handling this next.